`pymupdf` tests are sad. Converting to draft:

```
python3.12-pymupdf> FAILED tests/test_pagedelete.py::test_deletion - TypeError: pdf_rearrange_pages2() missing 1 required positional argument: '...
python3.12-pymupdf> FAILED tests/test_pagedelete.py::test_3150 - TypeError: pdf_rearrange_pages2() missing 1 required positional argument: '...
python3.12-pymupdf> FAILED tests/test_textextract.py::test_3705 - TypeError: pdf_rearrange_pages2() missing 1 required positional argument: '...
```

--- a/pkgs/development/python-modules/pymupdf/default.nix
+++ b/pkgs/development/python-modules/pymupdf/default.nix
@@ -4,6 +4,7 @@
   buildPythonPackage,
   pythonOlder,
   fetchFromGitHub,
+  fetchpatch,
   python,
 
   # build-system
@@ -41,7 +42,7 @@ let
 in
 buildPythonPackage rec {
   pname = "pymupdf";
-  version = "1.25.1";
+  version = "1.25.2";
   pyproject = true;
 
   disabled = pythonOlder "3.7";
@@ -50,8 +51,18 @@ buildPythonPackage rec {
     owner = "pymupdf";
     repo = "PyMuPDF";
     tag = version;
-    hash = "sha256-kdu8CuQJ5+h8+PS66acWEfcttgALiD+JBoWWyGtjBzs=";
+    hash = "sha256-6XbHQ8PE9IF0kngUhYkFSGjwgt+r+19v+PeDAQin2Ko=";
   };
+
+  patches = [
+    # Fix build against mupdf-1.25.3:
+    #   https://github.com/pymupdf/PyMuPDF/pull/4248
+    (fetchpatch {
+      name = "mupdf-1.25.3.patch";
+      url = "https://github.com/pymupdf/PyMuPDF/commit/f42ef85058fee087d3f5e565f34a7657aad11240.patch";
+      hash = "sha256-X5JF8nPLj4uubdEdvUJ5aEf0yZkW+ks99pzua0vCrZc=";
+    })
+  ];
 
   # swig is not wrapped as Python package
   # libclang calls itself just clang in wheel metadata

